_studio/enctools/aenc/include/aenc.h

```diff
@@ -0,0 +1,78 @@
+// Copyright (c) 2020 Intel Corporation
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in all
+// copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+
+#ifndef __AENC_H__
+#define __AENC_H__
+#include "mfxstructures.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+    typedef struct {
+        mfxU32 FrameWidth;
+        mfxU32 FrameHeight;
+        mfxU32 SrcFrameWidth;
+        mfxU32 SrcFrameHeight;
+        mfxI32 Pitch;
+        mfxU32 ColorFormat; //FourCC, e.g. MFX_FOURCC_NV12
+        mfxU32 StrictIFrame;
+        mfxU32 GopPicSize; //GOP size, from I to I, excluding last I
+        mfxU32 MinGopSize; //for adaptive I
+        mfxU32 MaxGopSize; //for adaptive I, next condition should be satisfyed (MinGopSize < MaxGopSize - MaxMiniGopSize)
+        mfxU32 MaxIDRDist; //IDR distance in frames
+        mfxU32 MaxMiniGopSize;
+
+        //toolset
+        mfxU32 AGOP;
+        mfxU32 ALTR;
+        mfxU32 AREF;
+        mfxU32 APQ;
+    } AEncParam;
+
+    typedef struct {
+        mfxU32 POC;
+        mfxU32 SceneChanged;
```

Comment:
this is true / false right? why waste 32 bits?

Reply:
Yes, it is true false. Bool here or mfxU16 will still use 4 bytes due to alignment reasons.